walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Set<String> attendees = new HashSet<String>(request.getAttendees());
+    List<TimeRange> initalRanges = new ArrayList<TimeRange>();
+    List<TimeRange> results = new ArrayList<TimeRange>();
+    final int minMeetingTime = 15;
+    
+    for(int start = 0; start <= TimeRange.END_OF_DAY; start += minMeetingTime) {
+        if(start + minMeetingTime > TimeRange.END_OF_DAY) {
+            initalRanges.add(TimeRange.fromStartEnd(start, TimeRange.END_OF_DAY, false));
```

Comment:
fromStartEnd(start, TimeRange.END_OF_DAY, /* inclusive= */false)

for readability. i find with APIs that take in true/false, i always have to look up the documentation for what it means... unless it's documented in the code as above.